app/lib/task/backend.dart

```diff
@@ -488,9 +499,12 @@ class TaskBackend {
     }
   }
 
-  /// Garbage collect [PackageState] and results from old runtimeVersions.
+  /// Garbage collect [Task] and results from old runtimeVersions.
   Future<void> garbageCollect() async {
-    await _db.tasks.deleteBeforeGcRuntime();
+    await db.tasks
+        .where((task) => task.runtimeVersion < gcBeforeRuntimeVersion.asExpr)
+        .delete()
+        .execute();
```

Comment:
Nit: wouldn't the `delete()` call before the `where()` make more sense? It kind of make sense for the `select()` projection to be just before e.g. `stream()` lie a few lines below, but for `delete()`, I could imagine it above it. This feels like the `delete()` and `execute()` calls should be a single method call, but maybe that fades more with further familiarity of the library use.